src/rust/ide/lib/view/src/graph_editor/component/project_name.rs

```diff
@@ -0,0 +1,314 @@
+
+/// ProjectName's animations handlers.
+#[derive(Debug,Clone,CloneRef)]
+pub struct Animations {
+    highlight   : Animation<f32>,
+    positioning : Animation<Vector3<f32>>
+}
+
+impl Animations {
+    /// Create new animations handlers.
+    pub fn new(network:&frp::Network) -> Self {
+        let highlight   = Animation::<_>::new(&network);
+        let positioning = Animation::<_>::new(&network);
+        Self{highlight,positioning}
+    }
+}
+
+
+
+// ========================
+// === ProjectNameModel ===
+// ========================
+
+#[derive(Debug,Clone,CloneRef)]
+#[allow(missing_docs)]
+pub struct ProjectNameModel {
+    logger         : Logger,
+    animations     : Animations,
+    view           : component::ShapeView<shape::Shape>,
+    display_object : display::object::Instance,
+    text_field     : TextField,
+    project_name   : Rc<RefCell<String>>,
+    renamed_output : frp::Source<String>,
+}
+
+impl ProjectNameModel {
+    /// Create new ProjectNameModel.
+    pub fn new(world:&World,frp:&Frp) -> Self {
+        let scene                 = world.scene();
+        let logger                = Logger::new("ProjectName");
+        let display_object        = display::object::Instance::new(&logger);
+        let font                  = scene.fonts.get_or_load_embedded_font("DejaVuSansMono").unwrap();
+        let size                  = Vector2::new(600.0,100.0);
+        let base_color            = DARK_GRAY_TEXT_COLOR;
+        let text_size             = 16.0;
+        let text_field_properties = TextFieldProperties{base_color,font,size,text_size};
+        let text_field            = TextField::new(&world,text_field_properties);
+        let view_logger           = Logger::sub(&logger,"view_logger");
+        let view                  = component::ShapeView::<shape::Shape>::new(&view_logger,scene);
+        let project_name          = Rc::new(RefCell::new(UNKNOWN_PROJECT_NAME.to_string()));
+        let renamed_output        = frp.outputs.renamed.clone();
+        let animations            = Animations::new(&frp.network);
+        Self{logger,view,display_object,text_field,project_name,renamed_output
+            ,animations}.initialize()
+    }
+
+    fn setup_center_alignment(&self) {
+        let mut width = 0.0;
+        self.text_field.with_mut_content(|content| {
+            let mut line = content.line(0);
+            if line.len() > 0 {
+                width = line.get_char_x_position(line.len() - 1);
+            }
+        });
+        let offset = Vector3::new(-width/2.0,0.0,0.0);
+        let mut height = default();
+            self.text_field.with_content(|content| height = content.line_height);
+        self.view.shape.sprite.size.set(Vector2::new(width,height));
+        self.view.set_position(Vector3::new(0.0,-height/2.0,0.0));
+        self.animations.positioning.set_target_value(offset);
+    }
+
+    fn initialize(self) -> Self {
+        self.add_child(&self.text_field.display_object());
```

Comment:
never, ever use `.disaply_object()` explicitelly - I add this comment to almost all PRs here.

Reply:
@wdanilo I have explained this before. `TextField` doesn't implement `display::Object` because of how it was designed.

Reply:
I don't think this is the right explanation. It would be OK for me if in every place you would create TODOs, but there are no todos. Moreover, text field should define it. Add it there instead of such code here.

Reply:
Well, I think it is. The only way to implement it AFAIR is to completely refactor TextField - and you are doing it anyway.

Look at commented code https://github.com/enso-org/ide/blob/main/src/rust/ensogl/src/display/shape/text/text_field.rs#L511 To implement this "From" properly the TextField cannot be under the shared! macro. But as I said that involves much refactoring, which is worthless if you replace that textfield anyway.

Reply:
@farmaazon I'm not sure if you have read my whole comment or only the first sentence. I told I'm ok with leaving it IFF in every usage place there would be fixme comment in order not to forget updating it after new text field kicks in

Reply:
FIXME notes everywhear